payment/controllers.go

```
@@ -276,10 +233,69 @@ func GetOrder(service *Service) handlers.AppHandler {
 			status = http.StatusNotFound
 		}
 
+		if order != nil && !order.IsPaid() && order.IsStripePayable() {
+			type OrderWithStripeCheckoutSessionID struct {
+				*Order
+				StripeCheckoutSessionID string `json:"stripeCheckoutSessionId"`
+			}
+
+			s, err := service.Datastore.GetOrderMetadata(*orderID.UUID(), "stripeCheckoutSessionId")
+			stripeCheckoutSessionID := s[1 : len(s)-1]
+
+			if err != nil {
+				return handlers.WrapError(err, "Error retrieving stripeCheckoutSessionId", http.StatusInternalServerError)
+			}
+
+			orderWithStripeCheckoutSessionID := OrderWithStripeCheckoutSessionID{
+				Order:                   order,
+				StripeCheckoutSessionID: stripeCheckoutSessionID,
+			}
+
+			return handlers.RenderContent(r.Context(), orderWithStripeCheckoutSessionID, w, http.StatusOK)
+		}
+
 		return handlers.RenderContent(r.Context(), order, w, status)
 	})
 }
 
+// CancelOrder is the handler for cancelling an order
+func CancelOrder(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var orderID = new(inputs.ID)
+		if err := inputs.DecodeAndValidateString(context.Background(), orderID, chi.URLParam(r, "orderID")); err != nil {
+			return handlers.ValidationError(
+				"Error validating request url parameter",
+				map[string]interface{}{
+					"orderID": err.Error(),
+				},
+			)
+		}
+
+		stripeSubscriptionID, _ := service.Datastore.GetOrderMetadata(*orderID.UUID(), "stripeSubscriptionId")
```

don't swallow errors